scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -3825,14 +3844,7 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
       case KwCase() if ahead(token.is[Ident]) && dialect.allowEnums =>
         enumCaseDef(mods)
       case KwIf() if mods.size == 1 && mods.head.is[Mod.Inline] =>
-        expr() match {
-          case ifExpr: Term.If =>
-            ifExpr.setMods(mods)
-            ifExpr
-          case expr => syntaxError("Unexpected expressions after inline modifier", at = expr.pos)
-        }
-      case KwIf() if mods.size > 0 =>
-        syntaxError("If can only contain inline modifier", at = token.pos)
+        ifClause(isInline = true)
```

Comment:
I removed the additional case with non-inline mods - it will be reported anyway as error.